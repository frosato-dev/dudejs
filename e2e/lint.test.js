import fs from "fs";
import { execSync } from "child_process";
import { CLIENT_DIR } from "./constants";

const FILE_PATH = `${CLIENT_DIR}/src/toLint.js`;

beforeAll(() => {
  fs.writeFileSync(FILE_PATH, `const unused = 1`);
});

it("should pass", () => {
  let stdout;
  try {
    execSync("npx dukejs lint", { cwd: CLIENT_DIR });
  } catch (err) {
    stdout = err.stdout.toString();
  }
  expect(stdout).toBeDefined();
  expect(stdout).toMatch(/error  'unused' is assigned a value but never used/);
});
