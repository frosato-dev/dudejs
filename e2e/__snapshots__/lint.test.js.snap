// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should lint all .js files under \`src\` directory 1`] = `
"
/private/tmp/dudejs-client/src/toLint-1.js
  1:7  error  'unused' is assigned a value but never used  no-unused-vars

/private/tmp/dudejs-client/src/toLint-2.js
  1:1  error  Unexpected console statement  no-console

âœ– 2 problems (2 errors, 0 warnings)

"
`;
